src/main/java/klerer/citibike/lambda/CitiBikeRequestHandler.java

```diff
@@ -20,8 +20,9 @@ public CitiBikeResponse handleRequest(APIGatewayProxyRequestEvent event, Context
 
         CitiBikeServiceFactory factory = new CitiBikeServiceFactory();
         CitiBikeService service = factory.getService();
+        StationsCache stationsCache = new StationsCache();
```

Comment:
You need to only create this once.

When you create this every time handleRequest() is called it defeats the purpose of having a cache.